recipes/lzip/all/conanfile.py

```diff
@@ -1,102 +1,63 @@
+        if self.settings.os == "Windows" and self.settings.compiler != "gcc":
+            raise ConanInvalidConfiguration("Only gcc supported for windows builds")
 
     def package_id(self):
         del self.info.settings.compiler
-
-    def _detect_compilers(self):
-        tools.rmdir("detectdir")
-        tools.mkdir("detectdir")
-        with tools.chdir("detectdir"):
-            tools.save("CMakeLists.txt", textwrap.dedent("""\
-                cmake_minimum_required(VERSION 2.8)
-                project(test C CXX)
-                message(STATUS "CC=${CMAKE_C_COMPILER}")
-                message(STATUS "CXX=${CMAKE_CXX_COMPILER}")
-                file(WRITE cc.txt "${CMAKE_C_COMPILER}")
-                file(WRITE cxx.txt "${CMAKE_CXX_COMPILER}")
-                """))
-            CMake(self).configure(source_folder="detectdir", build_folder="detectdir")
-            cc = tools.load("cc.txt").strip()
-            cxx = tools.load("cxx.txt").strip()
-        return cc, cxx
+        del self.info.settings.build_type
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @contextlib.contextmanager
-    def _build_context(self):
-        env = {}
-        cc, cxx = self._detect_compilers()
-        if not tools.get_env("CC"):
-            env["CC"] = cc
-        if not tools.get_env("CXX"):
-            env["CXX"] = cxx
-        with tools.environment_append(env):
-            yield
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        conf_args = [
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
 
-    def package(self):
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            with tools.environment_append({"CONAN_CPU_COUNT": "1"}):
-                autotools.install()
+        if self.settings.os == "Windows":
+            # Change program extension to .exe when compiling for windows
+            replace_in_file(self, os.path.join(self.source_folder, "Makefile.in"), '$(INSTALL_PROGRAM) ./$(progname) \"$(DESTDIR)$(bindir)/$(progname)\"', '$(INSTALL_PROGRAM) ./$(progname).exe \"$(DESTDIR)$(bindir)/$(progname).exe\"')
```

Comment:
Any insight into this change? Has it been reported upstream?

Reply:
@RubenRBS I have moved the `replace_in_file(...)` into a proper patch file and reported it upstream to lzip-bug@nongnu.org.

The configure script will now by itself determine the executable extension and apply it instead of this `replace_in_file` if we're compiling to windows.

This is a better solution which can also be used upstream.

Regards
Klaus